app/src/main/java/com/glia/exampleapp/MainFragment.java

```diff
@@ -147,10 +148,13 @@ private void setupAuthButtonsVisibility() {
                 containerView.findViewById(R.id.initGliaWidgetsButton).setVisibility(View.VISIBLE);
                 containerView.findViewById(R.id.authenticationButton).setVisibility(View.GONE);
                 containerView.findViewById(R.id.deauthenticationButton).setVisibility(View.GONE);
+                containerView.findViewById(R.id.visitor_code_button).setVisibility(View.GONE);
```

Comment:
Maybe for testing purposes, we should leave the button as is. Then we can test improper setup

Reply:
@gugalo What's your opinion on this? Shall we hide the "Show visitor code" button until "Init Glia Widgets" is tapped (to prevent the example app from crash), or leave it always visible?

Reply:
Hmm, that is a hard question and I have no strong opinion ))

But it is a public app/repository so I think at the bare minimum it should not crash in any case. Though we can also probably catch the exception and just show some Toast message, in that case, to be able to test this case.

@Pelkar do you have a strong opinion regarding this? If not then it is usually the PR owner who makes the final decision as it is his responsibility in the end. If you do have a strong opinion then we can discuss it tomorrow on standup or in #tm-mobile slack channel

Reply:
I just think that removing this from the test app, this can leave a blind spot during testing. Wrapping it try/catch would be a good thing to prevent the app from crashing, but I think it should be possible to initialize incorrectly.

Reply:
Let me clarify:
1. What do you mean by incorrect initialization?
2. How can making the "Show visitor code" button always visible help with what you mean?

Reply:
I mean that we could open visitor code without initialization like the integrator might do by accident or by not following the documentation

Reply:
Okay. ✅ 
And how can making the "Show visitor code" button always visible help with the fact that integrator should call `GliaWidgets.init` before Visitor code view usage?

Reply:
My point is simple, in my mind our test application should be able to replicate every problem the user might run into while implementing the library. Although an argument against it would be that we already do the same thing with Authenticate. We could go either way and I am ok with going with your solution